File: Lab 1/src/echo.rb
=begin
  Echo Program
  by Stuart C. Shapiro
  To demonstrate the writing and running of a Ruby program.
  To run this program:
    Typy ruby echo.rb

  Repeats the following:
    print a numbered prompt;
    read something;
    echo 'I read --><whatever was read><--'
  until the thing read is the symbol bye.
=end

def echo()
  puts "Ruby Echo"
  puts "Type something to each prompt."
  puts "Type bye to quit.\n\n"
  n = 1
  while true
    print "#{n}: "
    input = gets.chomp
    if input == "bye"
      break
    end
    puts "I read -->#{input}<--"
    n += 1
  end
  puts "Good bye."
end

echo()

File: Lab 1/src/echo.py
#! /usr/bin/python

#  Echo Program
#  by Stuart C. Shapiro
#
#  To demonstrate the writing and running of a Python program.
#
#  Store this program in a file named echo.py
#
#  To run this program:
#    1. Type chmod 700 echo.py
#            to make echo.py executable
#    2. Type ./echo.py
#  Or you needn't include the "shebang" line at the top
#     and you needn't make echo.py executable
#     Just type python echo.py
#
#    Repeats the following:
#      print a numbered prompt;
#      read something;
#      echo 'I read --><whatever was read><--'
#    until the thing read is the symbol bye.

def echo():
    print "Python Echo\nType something to each prompt.\nType `bye' to quit.\n\n";
    n = 1
    while True:
        input = raw_input(str(n) + ":  ")
        if input == "bye": break
        print "I read -->" + input + "<--\n"
        n += 1
    print "Good bye.\n"

echo()

